### PR TITLE
Update dependency version for U10

### PR DESCRIPTION
**Pull request by TharmiganK**
## Examples
N/A

## Checklist
- [ ] ~Linked to an issue~
- [ ] ~Updated the changelog~
- [ ] ~Added tests~
- [ ] ~Updated the spec~
- [ ] ~Checked native-image compatibility~

### CODE REVIEW

File: ballerina/Dependencies.toml [hunk 1]
```diff
@@ -5,7 +5,7 @@
 
 [ballerina]
 dependencies-toml-version = "2"
-distribution-version = "2201.8.0"
+distribution-version = "2201.10.0-20240802-124900-0b8d2ff7"
```
Comment (TharmiganK) [inline ballerina/Dependencies.toml:8]:
```suggestion
distribution-version = "2201.10.0-20240803-215800-d7bdc125"
```

File: gradle.properties [hunk 2]
```diff
@@ -1,7 +1,7 @@
 org.gradle.caching=true
 group=io.ballerina.stdlib
 version=2.10.0-SNAPSHOT
-ballerinaLangVersion=2201.8.0
+ballerinaLangVersion=2201.10.0-20240802-124900-0b8d2ff7
```
Comment (TharmiganK) [inline gradle.properties:4]:
```suggestion
ballerinaLangVersion=2201.10.0-20240803-215800-d7bdc125
```

File: gradle.properties [hunk 3]
```diff
@@ -12,5 +12,5 @@ mockitoVersion=5.3.1
 
 stdlibIoVersion=1.6.0
 stdlibLogVersion=2.9.0
-observeVersion=1.2.0
-observeInternalVersion=1.2.0
+observeVersion=1.3.0-20240802-173700-3e9a47e
```
Comment (TharmiganK) [inline gradle.properties:15]:
```suggestion
observeVersion=1.3.0-20240805-104200-3d4e4dd
```

File: gradle.properties [hunk 4]
```diff
@@ -12,5 +12,5 @@ mockitoVersion=5.3.1
...
+observeInternalVersion=1.3.0-20240802-182300-e86d5b5
```
Comment (TharmiganK) [inline gradle.properties:16]:
```suggestion
observeInternalVersion=1.3.0-20240805-105300-b9d6198
```

**Review by TharmiganK — COMMENTED** [review]
Update dependency versions

**Review by lnash94 — APPROVED** [review]

**Pull request merged**